bundles/org.openhab.core.config.discovery.addon.usb/src/main/java/org/openhab/core/config/discovery/addon/usb/UsbAddonFinder.java

```diff
@@ -0,0 +1,186 @@
+
+    public static final String SERVICE_TYPE = SERVICE_TYPE_USB;
+    public static final String SERVICE_NAME = SERVICE_NAME_USB;
+
+    /*
+     * Supported 'match-property' names
+     */
+    public static final String PRODUCT = "product";
+    public static final String MANUFACTURER = "manufacturer";
+    public static final String CHIP_ID = "chipId";
+    public static final String REMOTE = "remote";
+
+    public static final Set<String> SUPPORTED_PROPERTIES = Set.of(PRODUCT, MANUFACTURER, CHIP_ID, REMOTE);
+
+    private final Logger logger = LoggerFactory.getLogger(UsbAddonFinder.class);
+    private final Set<UsbSerialDiscovery> usbSerialDiscoveries = new CopyOnWriteArraySet<>();
+    private final Map<Long, UsbSerialDeviceInformation> usbDeviceInformations = new ConcurrentHashMap<>();
+
+    @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
+    protected void addUsbSerialDiscovery(UsbSerialDiscovery usbSerialDiscovery) {
+        usbSerialDiscoveries.add(usbSerialDiscovery);
+        usbSerialDiscovery.registerDiscoveryListener(this);
+        usbSerialDiscovery.doSingleScan();
+    }
+
+    protected synchronized void removeUsbSerialDiscovery(UsbSerialDiscovery usbSerialDiscovery) {
+        usbSerialDiscovery.unregisterDiscoveryListener(this);
+        usbSerialDiscoveries.remove(usbSerialDiscovery);
+    }
+
+    @Override
+    public Set<AddonInfo> getSuggestedAddons() {
+        Set<AddonInfo> result = new HashSet<>();
+        for (AddonInfo candidate : addonCandidates) {
+            for (AddonDiscoveryMethod method : candidate.getDiscoveryMethods().stream()
+                    .filter(method -> SERVICE_TYPE.equals(method.getServiceType())).toList()) {
+                Map<String, Pattern> matchProperties = method.getMatchProperties().stream()
+                        .collect(Collectors.toMap(property -> property.getName(), property -> property.getPattern()));
+
+                Set<String> propertyNames = new HashSet<>(matchProperties.keySet());
+                propertyNames.removeAll(SUPPORTED_PROPERTIES);
+
+                if (!propertyNames.isEmpty()) {
+                    logger.warn("Add-on '{}' addon.xml file contains unsupported 'match-property' [{}]",
+                            candidate.getUID(), String.join(",", propertyNames));
+                    break;
+                }
+
+                logger.trace("Checking candidate: {}", candidate.getUID());
+                for (UsbSerialDeviceInformation device : usbDeviceInformations.values()) {
+                    logger.trace("Checking device: {}", device);
+
+                    if (propertyMatches(matchProperties, PRODUCT, device.getProduct())
+                            && propertyMatches(matchProperties, MANUFACTURER, device.getManufacturer())
+                            && propertyMatches(matchProperties, CHIP_ID,
+                                    toHexString(device.getVendorId(), device.getProductId()))
```

Comment:
Does this now also allows for supporting addon suggestions by other `UsbSerialDiscovery` implementations like ser2net (https://github.com/openhab/openhab-core/pull/3930#issuecomment-1866295251)?

Reply:
Indeed. This finder accepts discovered USB devices from all installed 'UsbSerialDiscovery' implementations. Currently that means the following..
- linuxsyfs (works on Linux only)
- ser2net (works for ip connected devices)
- windowsregistry [wip] (works on Windows only)
- javaxusb [wip] (works on all platforms, but with limited data i.e. only chipId)

Notes:
1. As @mherwege requested I added another 'remote' property to the UsbSerialDiscovery implementations, so that the suggestion finder can (say) reject a remotely (ser2net) device via the match-properties. See ZigBee example below..
2. This suggestion finder will process inputs from all UsbSerialDiscovery implementations; so on a Linux system (say) you will currently get duplicated USB device lists from linuxsyfs and javaxusb. However we could in principle suppress javaxusb on Linux resp. Windows where the other UsbSerialDiscovery implementation would provide more complete data.

Apropos https://github.com/openhab/openhab-core/pull/3930#issuecomment-1866295251 please look at the match-properties for (say) the ZigBee binding [here](https://github.com/openhab/org.openhab.binding.zigbee/blob/360000619924b4190b575fc25545220f19fbdb55/org.openhab.binding.zigbee/src/main/resources/OH-INF/addon/addon.xml) where you can see that we have two match-properties .. one for matching device description strings, and the other for matching chipIds.

Reply:
> so that the suggestion finder can (say) reject a remote (ser2net) device via the match-properties. See ZigBee example below.

Are you sure it's still unsupported? There was https://github.com/openhab/org.openhab.binding.zigbee/pull/661.

Reply:
> Are you sure it's still unsupported? 

I am not sure at all .. @mherwege said it ;) .. but I can change that of course.

Reply:
@wborn do you know if the Z-Wave binding now also supports ip remote sticks?

Reply:
Cool. I have removed the checks from the respective bindings' discovery-methods xml.

Reply:
I also find it cool that OH will be able to suggest add-ons for a USB sticks connected to other computers exposed via ser2net. :sunglasses: